lib/ecl/well_ts.cpp

```diff
@@ -100,41 +104,23 @@ static well_node_type * well_node_alloc( well_state_type * well_state) {
 }
 
 
-static UTIL_SAFE_CAST_FUNCTION( well_node , WELL_NODE_TYPE_ID )
-static UTIL_SAFE_CAST_FUNCTION_CONST( well_node , WELL_NODE_TYPE_ID )
-
-
 static void well_node_free( well_node_type * well_node ) {
   well_state_free( well_node->well_state );
-  free( well_node );
-}
-
-static void well_node_free__( void * arg ) {
-  well_node_type * node = well_node_safe_cast( arg );
-  well_node_free( node );
+  delete well_node;
 }
 
-static int well_node_time_cmp( const void * arg1 , const void * arg2) {
-  const well_node_type * node1 = well_node_safe_cast_const( arg1 );
-  const well_node_type * node2 = well_node_safe_cast_const( arg2 );
-
-  if (node1->sim_time < node2->sim_time)
-    return -1;
-  else if (node1->sim_time == node2->sim_time)
-    return 0;
-  else
-    return 1;
 
+static bool well_node_time_cmp( const well_node_type * node1 , const well_node_type * node2) {
```

Comment:
Since this now implements the `<` test it should be renamed `well_node_time_lt`.